src/uct/ib/mlx5/dv/ib_mlx5_dv.c

```diff
@@ -129,9 +107,11 @@ ucs_status_t uct_ib_mlx5_devx_create_qp(uct_ib_iface_t *iface,
     }
     UCT_IB_MLX5DV_SET(qpc, qpc, pm_state, UCT_IB_MLX5_QPC_PM_STATE_MIGRATED);
     UCT_IB_MLX5DV_SET(qpc, qpc, pd, dvpd.pdn);
-    UCT_IB_MLX5DV_SET(qpc, qpc, uar_page, uar->uar->page_id);
+    UCT_IB_MLX5DV_SET(qpc, qpc, uar_page, attr->uar->uar->page_id);
     ucs_assert((attr->super.srq == NULL) || (attr->super.srq_num != 0));
-    UCT_IB_MLX5DV_SET(qpc, qpc, rq_type, !!attr->super.srq_num);
+    UCT_IB_MLX5DV_SET(qpc, qpc, rq_type, (attr->super.srq_num > 0) ? 1 :
+                                         (attr->super.cap.max_recv_wr == 0) ? 3 :
```

Comment:
Maybe add some constants. Function of uct_ib_qp_attr_t maybe. 
From PRM:
rq_type 0x0: regular - (non zero size) rq
0x1: srq/rmp/xrc_srq/xrq
0x3: zero_size_rq - (no_rq)